site/content/contribute/webapp/build-component.md

```diff
@@ -114,60 +117,50 @@ export default connect(mapStateToProps, mapDispatchToProps)(ItemList);
 
 If the selectors and/or actions you need don't yet exist in Redux then you should go add those first by following the [guide to adding actions and selectors](/contribute/redux/actions/).
 
-Your `index.js` and `item_list.js` files will live together in an `item_list/` directory.
+Your `index.ts` and `item_list.ts` files will live together in an `item_list/` directory.
 
-## Implementing your Component
+## Implementing the Component
 
-With the props defined and, if necessary, the container built you're ready to implement your component. For the most part, implementing a component for the webapp is no different than building any other React component. That said, we do have a few rules to abide by:
-
-1. All components must extend `React.PureComponent`.
-2. All data for rendering must come from props.
-3. Local rendering information that does not affect the store may be kept in state.
+With the props defined and, if necessary, the container built, you're ready to implement the rest of your component. For the most part, implementing a component for the web app is no different than building any other React component. While older code tends to use class components which inherit from `React.PureComponent`, most newer code should use functional components.
 
 Our `ItemList` example might look something like this:
 
-```javascript
-export default class ItemList extends React.PureComponent {
-    static propTypes = {
+```tsx
+type Props = {
+    /**
+     * The title of the list
+     */
+    title?: string;
+    
+    /**
+     * An array of item components to display
+     */
+    items: ListItem[];
+    
+    actions: {
         /**
-        * Sets the title of the list
+        * An action to remove an item from the list
         */
-        title: PropTypes.string,
+        removeItem: (item: ListItem) => void;
+    };
+}
 
-        /**
-        * An array of item components to display
-        */
-        items: PropTypes.arrayOf(PropTypes.object).isRequired
-
-        actions: PropTypes.shape({
-            /**
-            * An action to remove an item from the list
-            */
-            removeItem: React.PropTypes.func.isRequired
-        }).isRequired
-    }
-
-    render() {
-        const items = [];
-
-        this.props.items.forEach((item) => {
-            items.push(
-                <Item
-                    item={item}
-                    onClick={this.props.actions.removeItem}
-                />
-            );
-        });
-
-        return (
-            <div className='backstage-header'>
-                <h1>
-                    {this.props.title}
-                </h1>
-                {items}
-            </div>
-        );
-    }
+export default function ItemList(props: Props) {
+    const title = this.props.title ? <h1>{this.props.title}</h1> : undefined;
```

Comment:
I've started favouring `undefined` over null since it makes typing cleaner in some situations like when typing function arguments since it avoids some awful type signatures like `Channel | undefined | null` that make me die a little inside. Ideally, JavaScript just wouldn't have two zero values for references, but this is what we're stuck with.

Ranting about JS aside, yeah, I can change this. Most people are probably more familiar with just using null everywhere, and that's better since this is an introductory doc
```suggestion
    const title = this.props.title ? <h1>{this.props.title}</h1> : null;
```

Reply:
haha, i feel ya ... there is a of discussion in the community at all times about the `null` vs `undefined` problem. I prefer to see it as if I define a variable manually I use `null`, since there is some intent in it.